pyiceberg/table/delete_file_index.py

```diff
@@ -0,0 +1,165 @@
+from typing import List, Dict, Optional, Any
+from bisect import bisect_left
+from pyiceberg.table import DataFile
+from pyiceberg.manifest import ManifestEntry, DataFileContent
+from pyiceberg.types import NestedField
+
+
+class EqualityDeleteFileWrapper:
+    def __init__(self, spec_id: int, manifest_entry: ManifestEntry, schema):
+        self.spec_id = spec_id
+        self.manifest_entry = manifest_entry
+        self.delete_file = manifest_entry.data_file
+        self.schema = schema
+        self.apply_sequence_number = (manifest_entry.sequence_number or 0) - 1
+
+    def equality_fields(self) -> List[NestedField]:
+        """Get equality fields for current delete file"""
+        fields = []
+        for field_id in self.delete_file.equality_ids or []:
+            field = self.schema.find_field(field_id)
+            if field:
+                fields.append(field)
+        return fields
+
+
+class EqualityDeletesGroup:
+    def __init__(self):
+        self._buffer: List[EqualityDeleteFileWrapper] = []
+        self._sorted: bool = False
+        self._seqs: Optional[List[int]] = None
+        self._files: Optional[List[EqualityDeleteFileWrapper]] = None
+
+    def add(self, spec_id: int, manifest_entry: ManifestEntry, schema):
```

Comment:
This API is not very intuitive, is there anything preventing us from having `add(EqualityDeleteFileWrapper)` directly?